### ISSUE TITLE
Hot Reload in Blazor not working with Authorization-Middleware

### ISSUE THREAD

**Issue by Knarzi** [dotnet/runtime#59481]
In my Blazor Backend I use the Authorization-Middleware in combination with JwtBearer. Without this two lines
```c#
app.UseAuthentication();
app.UseAuthorization();
```
hot reload works like it schould. The browser reloads directly and I get the following output from dotnet watch
```
watch : Received 1 from browser in [Count: 1, MessageType: Binary, EndOfMessage: True].
watch : Received successful apply from delta applier.
watch : Hot reload of changes succeeded.
watch : Hot reload change handled in 556,7192ms.
```

But when I add hte two lines from above to use the authorization middleware, I only get the option to restart the application with following output
```
watch : Received 0 from browser in [Count: 0, MessageType: Close, EndOfMessage: True].
watch : Received 0 from browser in [Count: 1, MessageType: Binary, EndOfMessage: True].
watch : Received failed apply from delta applier.
watch : Unable to handle changes using hot reload.
watch : Do you want to restart your app - Yes (y) / No (n) / Always (a) / Never (v)?
```

Is that supposed to be the case or is that a bug? In any case, it would not be nice if hot reloading would not work in combination with this middleware.

**Comment by Knarzi** [comment]
@lambdageek Okay, so I wasn't able to reproduce the error in a test project, I uploaded the project as a private Github repository and invited you. To run and reproduce the problem, follow the instructions in the `Readme.md`. If you have additional questions, feel free to ask.

**Comment by lambdageek** [comment]
@Knarzi I don't think I got the invitation from github.  I added a private repo https://github.com/lambdageek/contact-info with my contact info  - send me an email we'll figure it out.

**Comment by lambdageek** [comment]
@knarzi got the github repo invite.  I can reproduce the issue.  Investigating...

**Comment by lambdageek** [comment]
Looks like there's a problem with `assembly.GetCustomAttributesData()` when `assembly` is the `"Anonymously Hosted DynamicMethods Assembly"`. 

The hot reload agent can work around this by skipping this loop

https://github.com/dotnet/aspnetcore/blob/216c92b78bce31d5e81a70b589707ec2ae5ab21a/src/Components/WebAssembly/WebAssembly/src/HotReload/HotReloadAgent.cs#L65

if the assembly name matched `"Anonymously Hosted DynamicMethods Assembly"`, @pranavkm.

I will try to figure out where the mono bug is here.  We shoudl probably fix this for 6.0 @marek-safar

**Comment by lambdageek** [comment]
Console repro:
```
using System.Reflection;
using System.Reflection.Emit;

var dm = new DynamicMethod("fancyPants", typeof(void), new Type[]{});
var ilg = dm.GetILGenerator();

foreach (var assm in AppDomain.CurrentDomain.GetAssemblies()) {
    Console.WriteLine ("in {0}", assm);
    foreach (var cattr in assm.GetCustomAttributesData()) {
	Console.WriteLine ("    - {0}", cattr);
    }
}
```

compile with `dotnet publish --self-contained -r osx-x64 -p:UseMonoRuntime=true` run with `./bin/Debug/net6.0/osx-x64/publish/ca`

**Comment by lambdageek** [comment]
Verified that @knarzi sample works with `dotnet watch` in .NET 6 RC2.  I believe the underlying issue was fixed by #58201.

**Comment by Knarzi** [comment]
@lambdageek I tried out the latest rc2 daily build and can confirm, that hot reload works with my project.  Thanks for your help!
